nifi-nar-bundles/nifi-evtx-bundle/nifi-evtx-processors/src/main/java/org/apache/nifi/processors/evtx/ParseEvtx.java

```diff
@@ -0,0 +1,353 @@
+
+    private final FileHeaderFactory fileHeaderFactory;
+    private final MalformedChunkHandler malformedChunkHandler;
+    private final RootNodeHandler rootNodeHandler;
+    private final XMLStreamWriterFactory xmlStreamWriterFactory;
+    private final ResultProcessor resultProcessor;
+
+    public ParseEvtx() {
+        this(FileHeader::new, ParseEvtx::handleMalformedChunkException, XmlBxmlNodeVisitor::new,
+                ParseEvtx::createWriter, ParseEvtx::processResult);
+    }
+
+    @VisibleForTesting
+    ParseEvtx(FileHeaderFactory fileHeaderFactory, MalformedChunkHandler malformedChunkHandler, RootNodeHandler rootNodeHandler,
+              XMLStreamWriterFactory xmlStreamWriterFactory, ResultProcessor resultProcessor) {
+        this.fileHeaderFactory = fileHeaderFactory;
+        this.malformedChunkHandler = malformedChunkHandler;
+        this.rootNodeHandler = rootNodeHandler;
+        this.xmlStreamWriterFactory = xmlStreamWriterFactory;
+        this.resultProcessor = resultProcessor;
+    }
+
+    @VisibleForTesting
+    static XMLStreamWriter createWriter(OutputStream outputStream) throws XMLStreamException {
+        XMLStreamWriter xmlStreamWriter = XML_OUTPUT_FACTORY.createXMLStreamWriter(outputStream, "UTF-8");
+        xmlStreamWriter.writeStartDocument();
+        xmlStreamWriter.writeStartElement(EVENTS);
+        return xmlStreamWriter;
+    }
+
+    @VisibleForTesting
+    static void close(XMLStreamWriter xmlStreamWriter) throws XMLStreamException {
+        xmlStreamWriter.writeEndElement();
+        xmlStreamWriter.close();
+    }
+
+    @VisibleForTesting
+    static String getName(String basename, Object chunkNumber, Object recordNumber, String extension) {
+        StringBuilder stringBuilder = new StringBuilder(basename);
+        if (chunkNumber != null) {
+            stringBuilder.append("-chunk");
+            stringBuilder.append(chunkNumber);
+        }
+        if (recordNumber != null) {
+            stringBuilder.append("-record");
+            stringBuilder.append(recordNumber);
+        }
+        stringBuilder.append(extension);
+        return stringBuilder.toString();
+    }
+
+    @VisibleForTesting
+    static String getBasename(FlowFile flowFile, ComponentLog logger) {
+        String basename = flowFile.getAttribute(CoreAttributes.FILENAME.key());
+        if (basename.endsWith(EVTX_EXTENSION)) {
+            return basename.substring(0, basename.length() - EVTX_EXTENSION.length());
+        } else {
+            logger.warn("Trying to parse file without .evtx extension {} from flowfile {}", new Object[]{basename, flowFile});
+            return basename;
+        }
+    }
+
+    @VisibleForTesting
+    static void processResult(ProcessSession session, ComponentLog logger, FlowFile updated, Exception exception, String basename, ChunkHeader chunkHeader, Record record) {
```

Comment:
it doesn't need to be, all of those parameters are initialized in onTrigger so they aren't really scoped to the object, I figured it'd be easier to test this logic as a static method